doc/fluid/design/data_pipeline/data_pipeline.md

```diff
@@ -0,0 +1,92 @@
+# Design Doc: Fluid Data Pipeline
+
+This document is about how Fluid training and inference programs reads data.
+
+## Standard Data Format
+
+### Case 1: Data from Files
+
+Consider a Fluid trianing program, `resnet50.py`, needs to read data from disk:
+
+```bash
+cat data | python resnet50.py
+```
+
+The fact that the who collect the data might not be the same person who wrote `resnet50.py` inspires that
```

Comment:
This line can be re-phrased as follows:

Since the person who collects the data might be different from the person who wrote resnet50.py:

Reply:
Done